docs/src/filters/writing_custom_filters.md

```diff
@@ -38,109 +38,91 @@ These components come together to form the [filter chain].
-
-   const NAME: &str = "greet.v1";
-
-   // This creates adds an associated const named `FILTER_NAME` that points
-   // to `"greet.v1"`.
-   struct Greet;
-
-   impl Filter for Greet {
-       fn read(&self, mut ctx: ReadContext) -> Option<ReadResponse> {
-           ctx.contents.splice(0..0, String::from("Hello ").into_bytes());
-           Some(ctx.into())
-       }
-       fn write(&self, mut ctx: WriteContext) -> Option<WriteResponse> {
-           ctx.contents.splice(0..0, String::from("Goodbye ").into_bytes());
-           Some(ctx.into())
-       }
-   }
-   ```
-
-   Next, we implement a [FilterFactory] for it and give it a name:
-
-   ```rust
-   // src/main.rs
-   # const NAME: &str = "greet.v1";
-   # struct Greet;
-   # impl Filter for Greet {}
-   # use quilkin::filters::Filter;
-   use quilkin::filters::{CreateFilterArgs, Error, FilterFactory};
-
-   struct GreetFilterFactory;
-   impl FilterFactory for GreetFilterFactory {
-       fn name(&self) -> &'static str {
-           // We provide the name of filter that we defined earlier.
-           NAME
-       }
-       fn create_filter(&self, _: CreateFilterArgs) -> Result<Box<dyn Filter>, Error> {
-           Ok(Box::new(Greet))
-       }
-   }
-   ```
-
-1. **Start the proxy**
-
-   We can run the proxy in the exact manner as the default Quilkin binary using the [run][runner::run] function, passing in our custom [FilterFactory].
-   Lets add a main function that does that. Quilkin relies on the [Tokio] async runtime so we need to import that crate and wrap our main function with it.
-
-   Add Tokio as a dependency in `Cargo.toml`.
-   ```toml
-   [dependencies]
-   quilkin = "0.1.0-dev"
-   tokio = { version = "1", features = ["full"]}
-   ```
-
-   Add a main function that starts the proxy.
-   ```rust, no_run
-   // src/main.rs
-   # use quilkin::filters::{CreateFilterArgs, Filter, Error, FilterFactory};
-   # struct GreetFilterFactory;
-   # impl FilterFactory for GreetFilterFactory {
-   #     fn name(&self) -> &'static str {
-   #         "greet.v1"
-   #     }
-   #     fn create_filter(&self, _: CreateFilterArgs) -> Result<Box<dyn Filter>, Error> {
-   #         unimplemented!()
-   #     }
-   # }
-   use quilkin::{filters::DynFilterFactory, run};
-
-   #[tokio::main]
-   async fn main() {
-       run(vec![Box::new(GreetFilterFactory) as DynFilterFactory]).await.unwrap();
-   }
-   ```
+#### 1. Import the Quilkin crate
+
+```bash
+# Start with a new crate
+cargo new --bin quilkin-filter-example
+```
+Add Quilkin as a dependency in `Cargo.toml`.
+```toml
+[dependencies]
+quilkin = "0.2.0"
+```
+
+#### 2. Implement the filter traits
+
+It's not terribly important what the filter in this example does so let's write a `Greet` filter that appends `Hello` to every packet in one direction and `Goodbye` to packets in the opposite direction.
+
+We start with the [Filter] implementation
+
+```rust,no_run,noplayground
+struct Greet;
+
+impl Filter for Greet {
```

Comment:
I couldn't include this, so just copied it inline. In theory it could go out of date, but I couldn't think of a better way.

Reply:
This should no longer apply, as now it matches how it was before.